util/mcf_dict_util.py

```diff
@@ -0,0 +1,547 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Utility function to read, edit and write MCF files.
+MCF files are read into dict like objects which can be easily modified and written back.
+The format of dict would be as follows:
+{
+    'p1': {
+        'value': '<value of the prop>',
+        'namespace': '<namespace of the value>'
+    },
+    'p2': {
+        'value': '<value of the prop>',
+        'namespace': '<namespace of the value>',
+        (present only when [X Y ..] like complex values are present)
+        'complexValue': [
+            'v1',
+            'v2'
+            ..
+        ]
+    }
+    ...
+}
+Functions are provided to read or write single/multiple files.
+When multiple files are read, a dict object is created with a list similar to
+    above described dict like objects corresponding to each file.
+Functions to edit mcf files include:
+- Change property name.
+- Change value of given property.
+- Change namespace.
+- Check existence of dcids in DC.
+- Check existence of dcid in other set of nodes.
+- Drop a list of nodes with given list dcid.
+
+Caveats:
+- Text values with ':' within them, eg: prop: "my value : with colon" 
+    would not be easily accessible in dict form. It would be treated as multiple value.
+- All comments are moved to the top of the node if sort_keys option is set with write to mcf.
```

Comment:
comments always seem to be moved to the top of the node even if sort_keys is False.